src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/ReconstructWork.hpp

```diff
@@ -36,34 +36,49 @@ class EquationOfState;
 
 namespace grmhd::ValenciaDivClean::fd {
 /*!
- * \brief Reconstructs \f$\rho, p, u_i, B^i\f$, and \f$\Phi\f$, then computes
- * the Lorentz factor, upper spatial velocity, specific internal energy,
- * specific enthalpy, and the conserved variables. All results are written into
- * `vars_on_lower_face` and `vars_on_upper_face`.
+ * \brief Reconstructs the `PrimTagsForReconstruction` (usually
+ * \f$\rho, p, Wv^i, B^i\f$, and \f$\Phi\f$), and if `compute_conservatives` is
+ * true computes the Lorentz factor, upper spatial velocity, specific internal
+ * energy, specific enthalpy, and the conserved variables.
+ *
+ * All results are written into `vars_on_lower_face` and `vars_on_upper_face`.
+ *
+ * The reason the `PrimTagsForReconstruction` can be specified separately is
+ * because some variables might need separate reconstruction methods from
+ * others, e.g. to guarantee the reconstructed solution is positive.
  */
-template <typename PrimsTags, typename TagsList, size_t ThermodynamicDim,
-          typename F>
+template <typename PrimTagsForReconstruction, typename PrimsTagsVolume,
+          typename TagsList, size_t ThermodynamicDim, typename F,
+          typename PrimsTagsSentByNeighbor>
 void reconstruct_prims_work(
     gsl::not_null<std::array<Variables<TagsList>, 3>*> vars_on_lower_face,
     gsl::not_null<std::array<Variables<TagsList>, 3>*> vars_on_upper_face,
-    const F& reconstruct, const Variables<PrimsTags>& volume_prims,
+    const F& reconstruct, const Variables<PrimsTagsVolume>& volume_prims,
     const EquationsOfState::EquationOfState<true, ThermodynamicDim>& eos,
     const Element<3>& element,
     const FixedHashMap<
         maximum_number_of_neighbors(3), std::pair<Direction<3>, ElementId<3>>,
-        std::vector<double>,
+        Variables<PrimsTagsSentByNeighbor>,
         boost::hash<std::pair<Direction<3>, ElementId<3>>>>& neighbor_data,
-    const Mesh<3>& subcell_mesh, size_t ghost_zone_size);
+    const Mesh<3>& subcell_mesh, size_t ghost_zone_size,
+    bool compute_conservatives);
 
 /*!
- * \brief Reconstructs the mass density, velocity, and pressure, then computes
- * the specific internal energy and conserved variables. All results are written
- * into `vars_on_face`.
+ * \brief Reconstructs the `PrimTagsForReconstruction` and if
+ * `compute_conservatives` is `true` computes the specific internal energy and
```

Comment:
Please mention Lorentz factor, `v^i`, specific internal energy (same as above)

Reply:
I've attempted to do the right thing :P Let me know if I got it wrong :D